community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -328,7 +328,7 @@ public long nodeGetFromUniqueIndexSeek(
         }
 
         // If we find the node - hold a shared lock. If we don't find a node - hold an exclusive lock.
-        Locks.Client locks = state.locks();
+        Locks.Client locks = state.locks().implicit();
```

Comment:
it looks like implicit locks, as they are defined now, are not suitable for this case